Sync with 5.14

skmedix opened the pull request.

embeddedt committed:
Add option to adjust thread priorities (#162)

Client thread given default priority + 2
Server thread given min priority + 2
Chunk builder threads given min priority + 1

Number of chunk builders is also adjusted to match Sodium
defaults on modern versions

(cherry picked from commit f73ce5bdc3178f7bf0cb9d9b62d0315a09cc63ad)

Rongmario committed:
Update to use RetroFuturaGradle instead of ForgeGradle 2.3

(cherry picked from commit 5e6e40aaf0f17a5997a73158a492550a90417043)

Rongmario committed:
Deobf baubles + IE for dev

(cherry picked from commit 12628bf2af2ef8800806687806a6190e7e5d98a6)

Rongmario committed:
Fixed import + embed configuration not working

(cherry picked from commit c0f3dbbe6e712db3371cb0e5d79abc7e081d23d1)

Rongmario committed:
Mixin AP + RFG update

(cherry picked from commit 810bf5b7f36c502dc3c540daa87182e757c3f047)

Rongmario committed:
IntelliJ moment

(cherry picked from commit 2dfea742faf2003ebd60fe57159c8229dc71ac3a)

Rongmario committed:
Fixes #125

(cherry picked from commit c17983c1366e08cd9c57be2e4e040dbaec89be5a)

Rongmario committed:
Fixes #135

(cherry picked from commit bbb4a199368d06ef61f007760a611aaf923ce94c)

Rongmario committed:
Bump to 5.7

(cherry picked from commit db6cce7ee39c4043ad6f4846aaa1de7308107696)

embeddedt committed:
Re-add clamping logic from vanilla for render builder count (#167)

Fixes #166

(cherry picked from commit ce40e7390319b32ff3b7fbd197aabc6c71d54bbd)

Rongmario committed:
Bump to 5.8

(cherry picked from commit cacff27fdacf8bd23df3dc18e64b48e4b7318e53)

embeddedt committed:
Revamp releaseSpriteFramesCache (#168)

* Now loads/unloads frame data dynamically for ALL `TextureAtlasSprites`
* Data not referenced for 20 consecutive client ticks is unloaded
* Synchronize some methods in `FramesTextureData` to help prevent crashes

(cherry picked from commit 8b423982e45039d9cc51ffa7ce2114d054d5dd18)

Rongmario committed:
Bump to 5.9

(cherry picked from commit 4021bf0649e910f5c38e7cf29cecb6daa319f1fa)

embeddedt committed:
Fix FramesTextureData not actually generating mipmaps (#170)

Should be the actual fix for #60, #169

(cherry picked from commit 1a3a56c22b07bc7bd6ae8fe0f055758f13de0671)

Rongmario committed:
Bump to 5.10

(cherry picked from commit 726645a17065f9db27a48d138cc552308bf9c1fd)

ISurrealI committed:
Fix Thermal Expansion machine item texture bug (#172)

A weird bug that happens because of face being null. Returning down or any other facing doesn't change the rendering. So i guess that's a fix?

(cherry picked from commit a1e25e1b141778f53185d6ea37d54ac64a8be5b7)

Rongmario committed:
Fixed incompatibility with MixinBooter's crash section for mixins

(cherry picked from commit 2241c2b28f2c41143960c251b904897a01552697)

Rongmario committed:
Bump to 5.11

(cherry picked from commit aceebf51799493f1f75a81637c2d36612cc75805)

Rongmario committed:
Alleviate some misery 8u51 users are having by updating CA Certs

(cherry picked from commit d44e45fd386b84c76e29caa78ea6aac7413abbd2)

Rongmario committed:
Bump to 5.12

(cherry picked from commit b4fbed33fbb6ca6dbf541ea58a59def3046add85)

embeddedt committed:
Optimize iteration over ticking FramesTextureData instances (#175)

(cherry picked from commit 431a044875cd86898cb8f4ee63fcb0c559f64999)

Desoroxxx committed:
Fix `player.setInvulnerable(true);` not working (#173)

* Fix `player.setInvulnerable(true);` not working properly

(cherry picked from commit 4ed45d34a0ecfb2a7fa7bec64ce3753930e8a47a)

Rongmario committed:
Bump to 5.13

(cherry picked from commit 2b9d89b5816c0ecd9c22413a448c79946d0101d8)

embeddedt committed:
Optimize updateAnimations iteration (#182)

* Use LinkedHashSet so iteration time is dependent on size instead of capacity
* Clear whole set at once instead of removing items one by one
* Iterate listAnimatedSprites instead of mapUploadedSprites

Fixes #180

(cherry picked from commit dd6e234f7b98e30d8c7973d3ac5ab308f8fc0544)

Rongmario committed:
Magma compatibility (#184)

* Magma compatibility + add logging

(cherry picked from commit 2df63fb3f92a1712c5432d2256db90aca8e42d83)

Rongmario committed:
Bump to 5.14

(cherry picked from commit 16a9d12b277063605e49c5a0bc7ac833ec4122f7)

skmedix commented:
Friendly ping @mirrorcult

mirrorcult commented:
ah mb, missed this, will merge & release later today or tomorrow

BinBashBanana commented:
Yes please!

BinBashBanana commented:
Build for anyone who wants it, github won't let me upload jar files so you'll need to rename this.
[normalasm-5.14.jar.zip](https://github.com/mirrorcult/NormalASM/files/12270119/normalasm-5.14.jar.zip)

MojangPlsFix committed:
Config "threadPriorityFix" now disables mixin as well (#199)

(cherry picked from commit b41880b5b3441d67790b410fc1b5b3355c5218db)

TheComputerizer committed:
Fix a crash with SGCraft passing a null TextureAtlasSprite through its renderer #137 (#211)

(cherry picked from commit 5575cee2607430f5f43668ac1ecabae7810d5fbb)

Rongmario committed:
Fixes #189

(cherry picked from commit 36ed50ee72e2b5215cd860051fe8208b937b93e4)

Rongmario committed:
Bump to 5.15

(cherry picked from commit 6e6c4184d510785c3ab36ce76c105a18b2e67200)

TheComputerizer committed:
Fix makeEventsSingletons breaking capabilities (#213)

* Fix capability maps being cleared before dispatchers were made

(cherry picked from commit 7c4f2200f0eb7fd463c1521283725af8180bcfd0)

Rongmario committed:
Typo, use the uncasted NeighborNotifyEvent

(cherry picked from commit 1aeee85c73e8b7c198b0c4e3f90f1ff80a2e8bda)

Rongmario committed:
Refactored some of `IRefreshEvent` + avoid World objs from leaking

(cherry picked from commit e6bc5591e036e33295630a595bcac7de64cdd3b0)

Rongmario committed:
Implement BlockEvent stuff in NeighborNotifyEvent for now

(cherry picked from commit ff110d56c8d9bba57e370ce874aaa83b2cfdc3bc)

Rongmario committed:
Bump to 5.16

(cherry picked from commit 00622a1b29bb16b3ad8012d602418e5133a44111)

BinBashBanana commented:
bump

Rongmario committed:
Fixes #218

(cherry picked from commit beedb68e2dcc143664bf20c17fb35998875ce380)

Rongmario committed:
Bump to 5.17

(cherry picked from commit 000f6362b3a185b6a931ec8d0afc1102cf55c628)

skmedix commented:
I will update my fork in few days

embeddedt committed:
Move BufferBuilder texture detection logic to a public method (#219)

This makes it possible to call that logic from another mixin without reimplementing the whole thing

(cherry picked from commit f12b0dab638d16c9f7c0e89156b37162f6850728)

Rongmario committed:
Bump to 5.18

(cherry picked from commit 77716031cc5f8f35ce2c03c4db1e552d4624bd3e)

skmedix commented:
https://github.com/skmedix/NormalASM/releases/tag/5.18

Rongmario committed:
Fixes #239 that was introduced in embeddedt@8b42398

(cherry picked from commit 260bbd6203569da608fba257beb5866e74783cfe)

Rongmario committed:
Bump to 5.19

(cherry picked from commit bce972984659ed9135d5b7e40a71e543f4671725)

skmedix commented:
https://github.com/skmedix/NormalASM/releases/tag/5.19

Rongmario committed:
Set squashBakedQuads to false if any sodium ports are installed

(cherry picked from commit 7f64ad7768b14ee66cbe749fba1717e2fd269f67)

Rongmario committed:
Update cacerts

(cherry picked from commit 8f76f3016b1945e6dbd5747e0c74bf01652139e5)

Rongmario committed:
Allow normalasm to load earlier (stupid forge gimmick)

(cherry picked from commit 9176ad8b8529224121d588f487b718f95a911ddc)

Rongmario committed:
Bump to 5.20, I swear v6 is coming soon!

(cherry picked from commit 4ac51b1505dad41fb44971d32f85c2f9a556f556)

CaliforniaDemise committed:
fix optimizeNBTTagCompoundBackingMap option (#269)

* Compound NBT will replace array map with hash map if maps size reaches threshold.
* Set default threshold to 5.
* Implement TagMap structure for better readability.
* Respect threshold on clear and putAll method calls.

(cherry picked from commit 99a6753f76f110e4408b64c11aea4e4afb2cd248)

Rongmario committed:
Bump to 5.21, we stray further away from v6

(cherry picked from commit d82c2bc2845bfa25384a01e7ad03dd09e08dd156)

Rongmario committed:
Fixes #270, fixes #271

(cherry picked from commit 2bd2552e301866372466345b055694ca0d3fc598)

Rongmario committed:
Bump to 5.22

(cherry picked from commit 5df6f76dee96837423232b7ec3bbc9298e49c753)

Rongmario committed:
Disable on demand animations when Optifine is installed

(cherry picked from commit 3bbcfb86b85cfc2fb609093ab9f9b3a4ee5e7225)

Rongmario committed:
Bump to 5.23

(cherry picked from commit 09c0b1e3f3ead5e5db8a40448aee49ca41a26cea)
